pkg/bzz/address.go

```diff
@@ -105,7 +105,19 @@ func generateSignData(underlay, overlay []byte, networkID uint64) []byte {
 }
 
 func (a *Address) Equal(b *Address) bool {
-	return a.Overlay.Equal(b.Overlay) && a.Underlay.Equal(b.Underlay) && bytes.Equal(a.Signature, b.Signature) && bytes.Equal(a.Transaction, b.Transaction)
+	if a == nil || b == nil {
+		return a == b
+	}
+
+	return a.Overlay.Equal(b.Overlay) && multiaddrEqual(a.Underlay, b.Underlay) && bytes.Equal(a.Signature, b.Signature) && bytes.Equal(a.Transaction, b.Transaction)
+}
+
+func multiaddrEqual(a, b ma.Multiaddr) bool {
```

Comment:
Shouldn't that go into the utilities as well?

Reply:
`Multiaddr` it's not part of `bee` package.